bigquery/snippets/snippet.go

```diff
@@ -1136,6 +1474,134 @@ func importJSONAutodetectSchema(client *bigquery.Client, datasetID, tableID stri
 	return nil
 }
 
+func importJSONWithCMEK(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_load_table_gcs_json_cmek]
+	gcsRef := bigquery.NewGCSReference("gs://cloud-samples-data/bigquery/us-states/us-states.json")
+	gcsRef.SourceFormat = bigquery.JSON
+	gcsRef.AutoDetect = true
+	loader := client.Dataset(datasetID).Table(tableID).LoaderFrom(gcsRef)
+	loader.WriteDisposition = bigquery.WriteEmpty
+	loader.DestinationEncryptionConfig = &bigquery.EncryptionConfig{
+		// TODO: Replace this key with a key you have created in KMS.
+		KMSKeyName: "projects/cloud-samples-tests/locations/us-central1/keyRings/test/cryptoKeys/test",
+	}
+
+	job, err := loader.Run(ctx)
+	if err != nil {
+		return err
+	}
+	status, err := job.Wait(ctx)
+	if err != nil {
+		return err
+	}
+
+	if status.Err() != nil {
+		return fmt.Errorf("Job completed with error: %v", status.Err())
+	}
+
+	// [END bigquery_load_table_gcs_json_cmek]
+	return nil
+}
+
+func importORC(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_load_table_gcs_orc]
+	gcsRef := bigquery.NewGCSReference("gs://cloud-samples-data/bigquery/us-states/us-states.orc")
+	gcsRef.SourceFormat = bigquery.ORC
+	loader := client.Dataset(datasetID).Table(tableID).LoaderFrom(gcsRef)
+
+	job, err := loader.Run(ctx)
+	if err != nil {
+		return err
+	}
+	status, err := job.Wait(ctx)
+	if err != nil {
+		return err
+	}
+
+	if status.Err() != nil {
+		return fmt.Errorf("Job completed with error: %v", status.Err())
+	}
+	// [END bigquery_load_table_gcs_orc]
+	return nil
+}
+
+func importORCTruncate(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_load_table_gcs_orc_truncate]
```

Comment:
Do we want to combo all the `_append` and `_truncate` samples like we did in Python to save on system test time?

https://github.com/GoogleCloudPlatform/google-cloud-python/blob/b176b888d56651689bea8768d39e57e518660ce7/docs/bigquery/snippets.py#L1543-L1559

Reply:
My primary question is how are people consuming the samples?  If they're consuming from the docs, then the sample is straightforward.  If they're using the "view on github", then tag interleaving within a method like that makes for somewhat confusing samples.

Reply:
Our hypothesis is copy-paste from cloud.google.com, but it's a good question.